### PR TITLE
Make MAX_POOL_STACKS a generic const

### PR DESCRIPTION
**Pull request by fbryden**

### CODE REVIEW

File: regex-automata/src/meta/regex.rs [hunk 1]
```diff
@@ -32,7 +32,7 @@ use crate::{
 type CachePool = Pool<Cache, CachePoolFn>;
 
 /// Same as above, but for the guard returned by a pool.
-type CachePoolGuard<'a> = PoolGuard<'a, Cache, CachePoolFn>;
+type CachePoolGuard<'a> = PoolGuard<'a, Cache, CachePoolFn, 8>;
```
Comment (fbryden) [inline regex-automata/src/meta/regex.rs]: I got lazy to go further down the chain - so I left a hard-coded const here

Comment (fuchsnj) [reply regex-automata/src/meta/regex.rs]: nit: You could introduce a new type, such as `type CachePoolGuardWithStacks<'a, const STACKS: usize> = PoolGuard<'a, Cache, CachePoolFn, STACKS>;` so you don't have to adjust it everywhere, but a generic alias is still available.

**Review by fuchsnj — APPROVED** [review]
1 small nit, not that important though

File: regex-automata/src/meta/regex.rs (hunk 1 above)
Comment (fbryden) [reply regex-automata/src/meta/regex.rs]: Good idea :)
(and I understand it fully :p)

**Pull request merged**